player/lua/select.lua

```diff
@@ -353,6 +360,77 @@ mp.add_key_binding(nil, "select-audio-device", function ()
     })
 end)
 
+local function format_history_entry(entry)
+    local status
+    status, entry.time = pcall(os.date, options.history_date_format, entry.time)
+
+    if not status then
+        mp.msg.warn(entry.time)
+    end
+
+    return "(" .. entry.time .. ") " ..
+           (entry.title or select(2, utils.split_path(entry.path))) ..
+           " (" .. entry.path .. ")"
```

Comment:
filename + path is little bit long. It won't fit in most cases. In case there is no title, we could show only directory `select(1, utils.split_path(entry.path))`, filename is already redundant. It still won't fit, but may be bit better.